### ISSUE TITLE
Forbid to repeat instance variables on a class level

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#581]
# Rule request

## Thesis

https://github.com/amontalenti/elements-of-python-style/blob/master/README.md#dont-repeat-instance-labels-in-the-class

```python
# bad
class JSONWriter(object):
    handler = None
    def __init__(self, handler):
        self.handler = handler

# good
class JSONWriter(object):
    def __init__(self, handler):
        self.handler = handler
```

## Reasoning

This can cause errors and frustration.

**Comment by phoolish-philomath** [comment]
I can do this one.

**Comment by sobolevn** [comment]
@phoolish-philomath awesome! Please, ask me any question you have 🙂

**Comment by sobolevn** [comment]
Just to be clear, annotations are ok:

```python
class Some(object):
     field: int

    def __init__(self) -> None:
         self.field = 1
```

**Comment by sobolevn** [comment]
Well, actually I had a lot of thoughts about this one. And actually we don't need this rule.
Why? Because we can use `ClassVar` type. And this type is better than anything we can achieve. 
Why? Because it works with the subclasses. And we can only work in the context of a single class.

**Comment by sobolevn** [comment]
Nope, my logic is incorrect.

Some people might not use `ClassVar` for a mistake. Or they might not use `mypy`.
So, we need to protect them from this error.

**Comment by sobolevn** [comment]
Closed by https://github.com/wemake-services/wemake-python-styleguide/commit/eb727926e3eea2429fefe98b6315aabd78990117
